addon/scripts/background.js

```diff
@@ -211,30 +228,59 @@ function beforeSendHeaders(request, url) {
  */
 function beforeRequest(details, url) {
     // Clear vars if they haven't been used for a while
-    if (Date.now() - 2000 > timeSinceLastResp) {
+    if (VAR_RESET && Date.now() - VAR_RESET_MS > timeSinceLastResp) {
         resetVars();
     }
 
+    // Only sign tokens if config says so and the appropriate header was received previously
+    if (!DO_SIGN || !readySign) {
+        return false;
+    }
+
+    // Different signing methods based on configs
+    let xhrInfo;
+    switch (CONFIG_ID) {
```

Comment:
Why not "if CONFIG_ID != 1 { throw() }  doStuf() " ?

Reply:
If we want to introduce new configurations, then maybe the switch is better?

Reply:
personally I avoid adding things that complicate stuff but "may be useful later". but here that's just small details. All good - LGTM